lua/null-ls/builtins/diagnostics/commitlint.lua

```diff
@@ -1,69 +1,118 @@
-        diagnostic.line = 3
+    local line_start = line:sub(1, 3)
+    for _, v in pairs(symbols) do
+        if line_start == v then
+            return true
+        end
     end
 
-    return diagnostic
+    return false
 end
 
-local handle_commitlint_output = function(params)
-    if params.output and params.output.results then
-        local output = params.output.results[1]
+---From a message token, finds its severity
+---@param message string
+local function severity(message)
+    local severities = {
+        ["✖"] = h.diagnostics.severities.warning,
+        ["⚠"] = h.diagnostics.severities.error,
+        ["ⓘ"] = h.diagnostics.severities.hint,
+        ["⧗"] = h.diagnostics.severities.information,
+    }
+    for pattern, kind in pairs(severities) do
+        if pattern:sub(1, 3) == message:sub(1, 3) then
+            return kind
+        end
+    end
+end
 
-        local parser = h.diagnostics.from_json({
-            severities = {
-                [1] = h.diagnostics.severities.warning,
-                [2] = h.diagnostics.severities.error,
-            },
-        })
+---Lexes @commitlint/cli's raw output into single messages that can later be categorized by severity.
+---@param output string
+---@return string[]
+local function lex(output)
+    ---@type string[]: Non-empty string array.
+    local lines = vim.tbl_filter(function(s)
+        return s ~= "" and true or false
+    end, vim.split(output, "\n"))
 
-        local violations = {}
-        for _, violation in ipairs(output.errors) do
-            table.insert(violations, violation_to_diagnostic(violation))
+    ---@type string[]: Array of diagnostic messages.
+    local messages = {}
+    for i = 1, #lines do
+        local message = ""
+        local curr_line = lines[i]
+        if message_start(curr_line) then
+            message = message .. curr_line
+            --- Go to next line, but only do anything if we're on the same message, else, go back up
+            i = i + 1
+            while not message_start(lines[i]) and lines[i] ~= nil do
+                message = message
+                    .. "\n" -- keep message formatting
+                    .. lines[i]
+                i = i + 1
+            end
         end
 
-        for _, violation in ipairs(output.warnings) do
-            table.insert(violations, violation_to_diagnostic(violation))
-        end
+        table.insert(messages, message)
+    end
+
+    return messages
+end
 
-        return parser({ output = violations })
+---@param params {}
+---@param done function
+local function on_output(params, done)
+    if type(params.output) ~= "string" then
+        return done()
     end
+
+    local diagnostics = {}
+
+    local messages = lex(params.output)
+    for i = 1, #messages do
+        local message = messages[i]
+        local diagnostic = {
+            row = 1,
+            col = 1,
+            source = "commitlint",
+            message = message:match("%s%s%s(.*)"),
```

Comment:
```suggestion
            message = message:match("%s%s%s(.*)") or "",
```

This caused issues for me, since Neovim expects this to always be defined.

Reply:
Thanks for the compliment! I agree. I'm of the opinion short is always best, therefore maybe is best to keep using the previous `built-in` version.
I'll look into contributing a sensible, parseable CLI, output format to `@commitlint/cli`.

Thanks for the review @jose-elias-alvarez